### ISSUE TITLE
CentOS/RHEL 7.6 repo still only has version 0.7.13. 

### ISSUE THREAD

**Issue by dmillerc76** [openzfs/zfs#8830]
### System information

Type | Version/Name
 --- | --- 
Distribution Name	|  CentOS 7.6
Distribution Version	|  7.6
Linux Kernel	|  3.10.x
Architecture	|  X86_64
ZFS Version	| 0.8.0
SPL Version	| 

### Describe the problem you're observing
CentOS/RHEL 7.6 repo still only has version 0.7.13. Just a reminder to please package up 0.8.0 for RHEL/CentOS 7.6 when you have time. Thank you!
### Describe how to reproduce the problem
On systems that have the zfs.repo for CentOS 7.6 

yum clean all

yum list zfs

Available Packages
zfs.x86_64                                                          0.7.13-1.el7_6                                                          zfs-kmod

**Comment by behlendorf** [comment]
Due to the overall conservative philosophy of CentOS/RHEL we *do not* plan to update the main CentOS/RHEL 7.6 repository with packages for 0.8.0.  It will continue to be updated with 0.7.x point releases which primarily contain import bug fixes, but no new features.

The default version of ZFS will be updated to 0.8.x when we add a new repository for the next RHEL/CentOS 7.7 feature release (7.7).  I've updated the [wiki](https://github.com/zfsonlinux/zfs/wiki/RHEL-and-CentOS) accordingly to mention this.

For anyone who wants to install 0.8.0 today for CentOS/RHEL 7.6 it can be installed by enabling the `zfs-testing` repository.

**Comment by jasker5183** [comment]
If it's not ready for prime time just SAY SO. Why didn't you guys do this with 0.5 to 0.6 or 0.6 to 0.7? What is so special now that we have to wait until 7.7? At which point I'll never get upgraded because you guys always FORGET to update the zfs-release in the EPEL repo.

**Comment by mskarbek** [comment]
what is so special? a huge amount of new features released at once. usually, if somebody is using centos, he's doing so because of a specific release model that promotes stability including features set stability. there is nothing that will stop you from upgrading to 0.8.0 individually but do not assume that everyone is ready to switch version of one of the fundamental components of infrastructure in the middle of the centos release cycle.

**Comment by jasker5183** [comment]
This is not going to be another 0.7.7 is it? If that's the case I will gladly wait for 7.7 to drop...who am I kidding I have no choice but to wait I'm not installing zfs-testing on anything production.

**Comment by kpande** [comment]
so if you agree about not upgrading to "experimental" release of ZFS, there is no problem here. thank you for being understanding.

**Comment by jasker5183** [comment]
Oh I understand I think it's a really stupid decision...but I understand (I don't).

**Comment by kpande** [comment]
the thing is, several of the community members expressed concern yesterday that stable CentOS users _would_ be updated to 0.8.0. there is nothing preventing you from building your own 0.8 packages if  you think it is stupid to be on 0.7-release.

**Comment by jasker5183** [comment]
I completely understand their concern, mskarbek is correct there are a lot of new features being added with the new release. However if you guys did not think 0.8.0 was fully formed after 5 rc's then why release it as stable? 

I did not say that it was stupid to be on 0.7, I'm saying the decision made by the zfs group to not package 0.8.0 for CentOS 7.6 IS because here is what is going to happen:

1. CentOS 7.7 comes out and my system is automatically upgraded to it.
2. epel-release is upgraded as well but the zfs-release maintainers forgot to update their package.
3. The /etc/yum.repo.d/zfs.repo file still points to 7.6.
4. Unless I remember to manually download the zfs-release-7.7 I will continue to run the old version.

If people are concerned 0.8.0 is going to break something why not just have a zfs0.7 and a zfs0.8. Cause I know you guys like extra work /s. Yes I could just build this myself.

**Comment by behlendorf** [comment]
The concern isn't that 0.8 isn't ready.  It is.  It's that many consumers of CentOS / RHEL expect to receive advance notice of significant software updates so they have an opportunity to plan for how and when it's deployed to their systems.

**Comment by jasker5183** [comment]
Just be prepared to add a "Why is my CentOS box not upgrading to 0.8.0" to the frequently asked questions.

**Comment by ryanjaeb** [comment]
If I want to try 0.8, but avoid anything that's actually testing, is it just a matter of being careful about the versions I install from `zfs-testing`?  I see:

```shh
sudo yum --enablerepo=zfs-testing-kmod --showduplicates list zfs
```

```
Installed Packages
zfs.x86_64                    0.7.13-1.el7_6                    @zfs-kmod
Available Packages
zfs.x86_64                    0.7.12-1.el7_6                    zfs-kmod
zfs.x86_64                    0.7.13-1.el7_6                    zfs-kmod
zfs.x86_64                    0.8.0-rc3.el7                     zfs-testing-kmod
zfs.x86_64                    0.8.0-rc4.el7                     zfs-testing-kmod
zfs.x86_64                    0.8.0-rc5.el7                     zfs-testing-kmod
zfs.x86_64                    0.8.0-1.el7                       zfs-testing-kmod
```

So if I stick to `major.minor.patch` versions like `0.8.0`, is that actually a stable version, but published to the testing repository?

**Comment by behlendorf** [comment]
@ryanjaeb that's right.  The 0.8.0-1.el7 packages are the final tagged released version.  They've simply only been published to the zfs-testing repository for now.

**Comment by seamusharris** [comment]
I'm getting a different result when I run ``sudo yum --enablerepo=zfs-testing-kmod --showduplicates list zfs`` and can't see anything later than rc2.

I've tried removing and reinstalling as described at https://github.com/zfsonlinux/zfs/wiki/RHEL-and-CentOS with no luck.

It seems odd that the two RCs that are listed for me don't show up for @ryanjaeb.

```
sudo yum --enablerepo=zfs-testing-kmod --showduplicates list zfs

Available Packages
zfs.x86_64      0.7.12-1.el7_6              zfs
zfs.x86_64      0.7.13-1.el7_6              zfs
zfs.x86_64      0.8.0-rc1.el7.centos        zfs-testing-kmod
zfs.x86_64      0.8.0-rc2.el7               zfs-testing-kmod
```

**Comment by dmillerc76** [comment]
@seamusharris Make sure you do `yum clean all` first.

**Comment by seamusharris** [comment]
@dmillerc76 I had tried that but tried it again and it hasn't made any difference.

**Comment by seamusharris** [comment]
I'm  not sure why my yum commands weren't clearing the cache properly, but manually removing ``/var/cache/yum/x86_64/7/zfs*`` has worked and I can now see the newer RCs as well as the final release.

**Comment by xgiovio** [comment]
just asking. Is it  safe to use the latest 0.8.x on centos 7.6? Thank you

**Comment by behlendorf** [comment]
@xgiovio yes, you can safely use the latest 0.8.x on centos 7.6.

**Comment by llamafilm** [comment]
How can I tell which version are considered stable?  This is what I see in yum:
```
zfs.x86_64                0.8.0-rc3.el7                 zfs-testing-kmod
zfs.x86_64                0.8.0-rc4.el7                 zfs-testing-kmod
zfs.x86_64                0.8.0-rc5.el7                 zfs-testing-kmod
zfs.x86_64                0.8.0-1.el7                   zfs-testing-kmod
zfs.x86_64                0.8.1-1.el7                   zfs-testing-kmod
```
I understand rc = release candidate so I should avoid that.  Above @behlendorf says 0.8.0-1.el7 is the latest, but now there is 0.8.1-1.el7.

BTW, Red Hat 7.7 is now released.

**Comment by h1z1** [comment]
tldr As has been said, there is nothing stopping you from building 0.8 or rolling your own RPM.

You're confusing zfs being stable with the rpm release being stable, they are different things.  A bug with the RPM itself for example doesn't  change zfs in any way.

When you have Petabytes of data on the line, Enterprises don't just upgrade storage for good reason.  I can only imagine the critical environments ZFS is deployed in where proper test processes _are_ in place.  Understanding the system is more valuable then new features and yes that includes any regressions.   I'm more concerned with upstream kernels breaking zfs because yet another Intel CPU bug requires a kernel upgrade.

**Comment by koollman** [comment]
> Due to the overall conservative philosophy of CentOS/RHEL we _do not_ plan to update the main CentOS/RHEL 7.6 repository with packages for 0.8.0. It will continue to be updated with 0.7.x point releases which primarily contain import bug fixes, but no new features.
> 
> The default version of ZFS will be updated to 0.8.x when we add a new repository for the next RHEL/CentOS 7.7 feature release (7.7). I've updated the [wiki](https://github.com/zfsonlinux/zfs/wiki/RHEL-and-CentOS) accordingly to mention this.
> 
> For anyone who wants to install 0.8.0 today for CentOS/RHEL 7.6 it can be installed by enabling the `zfs-testing` repository.

Now that both rhel 7.7 and centos 7.7 are released, is zfs-testing repository still the way to go,
or will there be soon-ish a corresponding repo, and http://download.zfsonlinux.org/epel/zfs-release.el7_7.noarch.rpm ?

The  [wiki](https://github.com/zfsonlinux/zfs/wiki/RHEL-and-CentOS) still say "These packages should not be used on production systems" (for zfs-testing repo)

**Comment by tonyhutter** [comment]
@koollman the plan is to have Centos 7.7 packages for 0.8.2, which is going though testing right now (#9161) and should be out soon.

**Comment by snajpa** [comment]
@jasker5183 Well if you value uptime and availability, you really might not want to deploy 0.8.x just yet. The project leadership is right not to include it (yet) for EL users, I've too been burned by 0.8.x in multiple ways, it's going to take a while to stabilize this.

The changes are just too many, too much at once, too many combinations to QA on together.

Believe me, when it comes to storage systems, you might not want to be as agile as everywhere else... it's only, you know, your data at stake :))

**Comment by jasker5183** [comment]
I've waited a year and change to be rid of spl I can wait longer. Should they not have created a 7.7 repo and created a 0.7.13 package? CentOS 7.7 has been out for a while now.

**Comment by snajpa** [comment]
Well I hate to break it to you, but 0.8.0 uses SPL just as well as 0.7.x did. And it probably always will, it's just included in the zfs repo now ;-)

**Comment by jasker5183** [comment]
Guess I should have mentioned was talking about the spl kernel module being merged into zfs leading to (hopefully) no mismatched dkms problems upon update.

**Comment by jasker5183** [comment]
Looks like they created a 7.7 repo and file but you'll have to go into your zfs.repo and change 7.7.7 to 7.7 maybe they caught this already though. 0.8.2 installed fine so far so good.

**Comment by tonyhutter** [comment]
@jasker5183 yea, I just caught that and fixed it.

**Comment by shodanshok** [comment]
What if want to continue using zfs-0.7.x on newer RHEL/CentOS 7.x releases (eg: 7.7, 7.8, etc) instead of upgrading to zfs-0.8? Should we continue using the 7.6 repo? What about kmod compatibility? Thanks.

**Comment by gmelikov** [comment]
Looks like stale, I'll close it.
